### DISCUSSION TITLE
Customize payload response / partial payload response

### DISCUSSION THREAD

**Discussion by do-me** [qdrant/qdrant#1471] [Ideas]
### Partial payload response
I have a use case where I would like like to keep all my data (each with id, payload, vector) in qdrant but when retrieving data I'd like to define myself what exactly is returned. 

At the moment in Qdrant
- id is always returned (which is fine as it's small)
- payload can be toggled true/false
- vector can be toggled true/false

Instead of just switching the payload on/off, I'd like to submit a schema of the payload I'd like to be returned. 

### Example

My payload might contain these nodes:

`payload={'Name': 'Testname', 'ID': '12345', 'Name acronym': 'TSTS', 'Record Number': 987654321, 'Text': 'A very long text I do not want to return!'}`

But I'd only like to return the `Name acronym` for example. 

`([Record(id=82157, payload={'Name acronym': 'TSTS'}), ...])`

**Comment by azayarni** [comment]
with_payload parameter can be an array of fields to return.

https://qdrant.github.io/qdrant/redoc/index.html#tag/points/operation/search_points

> with_payload | (WithPayloadInterface (WithPayloadInterface (boolean) or Array of WithPayloadInterface (strings) or (PayloadSelector (PayloadSelectorInclude (object) or PayloadSelectorExclude (object))))) or (any or null)Select which payload to return with the response. Default: All
> -- | --

PS: I highly encourage you to join our Discord server to get responses in minutes instead of weeks ;) https://qdrant.to/discord
